### DISCUSSION TITLE
Append PATH on Windows

### DISCUSSION THREAD

**Discussion by phil-opp** [community/community#25119] [Actions]
<p>I’m trying to install <a href="https://scoop.sh/" rel="nofollow noopener">https://scoop.sh/</a>, but I fail to add it to the PATH environment variable for subsequent steps. I tried:</p>
<pre><code>set PATH=C:\Users\runneradmin\scoop\shims;%PATH%
</code></pre>
<p>and</p>
<pre><code>[Environment]::SetEnvironmentVariable("Path", $env:Path, [System.EnvironmentVariableTarget]::Machine)
</code></pre>
<p>(with a modified $env:Path).</p>
<p>When I print the PATH variable in the next step, it always remains unchanged. What’s the correct way to do this?</p>

**Comment by chrispat** [comment, marked as answer]
<p>Each step is run its own process space so environment changes do not automatically propgate.  If you want to update the path you can use the <code>add-path</code> command as documented <a href="https://help.github.com/en/articles/development-tools-for-github-actions#add-a-system-path-add-path" rel="nofollow noopener">https://help.github.com/en/articles/development-tools-for-github-actions#add-a-system-path-add-path</a>.  </p>

**Comment by phil-opp** [comment]
<p>Thanks a lot, this worked without problems!</p>
<p>I saw this page before but somehow thought that it only applied to the Node.js toolkit.</p>

**Comment by dickmao** [comment]
<p>The cited documentation suggests the modification is seen by all subsequent steps in the job.  So it would seem at odds with your assertion that each step is its own process.</p>
<p>I’ve not been able to get <code>::add-path::/path/to/dir</code> to augment PATH, either within the step or within the job.  Would appreciate sample yaml that augments the PATH variable, as the documentation offers none.</p>

**Comment by D4nte** [comment]
<p>It seems like the correct way on Windows is (contrary to the doc):</p>
<pre><code>- name: Add strawperryperl to the PATH to override the existing Perl installation so we can compile OpenSSL locally
        run: echo "##[add-path]C:\strawberry\c\bin;C:\strawberry\perl\site\bin;C:\strawberry\perl\bin"
        if: matrix.os == 'windows'
</code></pre>

**Reply by cagnulein** [reply]
Error: The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

**Reply by airtower-luna** [reply]
See here for the replacement: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path

Mind that the example there is using Unix shell syntax, you might have to translate it if you use powershell or cmd.

**Comment by dickmao** [comment]
<p>Ultimately had to monkey at the typescript level <a href="https://github.com/dickmao/setup-paths" rel="nofollow noopener">https://github.com/dickmao/setup-paths</a> to change PATH.</p>

**Comment by liudonghua123** [comment]
<p>Hi, Is there any documentation about this windows add-path style? From the offical docs, I couldn’t find it.</p>

**Comment by liudonghua123** [comment]
<blockquote>
<p><a class="mention" href="/d4nte">@d4nte</a> wrote:</p>
</blockquote>
<blockquote>
<p>It seems like the correct way on Windows is (contrary to the doc):</p>
</blockquote>
<blockquote>
<ul>
<li>name: Add strawperryperl to the PATH to override the existing Perl installation so we can compile OpenSSL locally<br>
run: echo “##[add-path]C:\strawberry\c\bin;C:\strawberry\perl\site\bin;C:\strawberry\perl\bin”<br>
if: matrix.os == ‘windows’</li>
</ul>
</blockquote>
<p>Hi, Is there any documentation about this windows add-path style? From the offical docs, I couldn’t find it.</p>

**Comment by evandrocoan** [comment]
<p>&gt; Each step is run its own process space so environment changes do not automatically propgate.  If you want to update the path you can use the <code>add-path</code> command as documented <a href="https://help.github.com/en/articles/development-tools-for-github-actions#add-a-system-path-add-path" rel="nofollow noopener">https://help.github.com/en/articles/development-tools-for-github-actions#add-a-system-path-add-path</a>.  </p>
<p>Your link as countless others out there for the GitHub actions documentation are all dead.</p>

**Comment by pawelsalawa** [comment]
<p>A little bit of googling and here it is: <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path</a></p>
<p>Works, at least at the moment of writing this.</p>
